### DISCUSSION TITLE
Problem about fu_udev_device_pread_full

### DISCUSSION THREAD

**Discussion by jyu99** [fwupd/fwupd#3309] [Q&A] [answered]
Hi 

I have problem about the fu_udev_device_pread_full function. in pixart_rf   code snippet as below 

```
/* skip the wrong report id ,and keep polling until result is correct */
while (g_timer_elapsed (timer, NULL) * 1000.f < FU_PXI_DEVICE_NOTIFY_TIMEOUT_MS) {
if (!fu_udev_device_pread_full (FU_UDEV_DEVICE (self),
				port, res, (FU_PXI_DEVICE_NOTTFY_RET_LEN + 1) - port,
				error))
	return FALSE;
if (res[0] == PXI_HID_DEV_OTA_INPUT_REPORT_ID)
	break;
}
```

But there is a problem in OTA processing. If device is keyboard ,user could use the keyboard and report the key during the OTA processing.  Once user report the media key(2 bytes) during the OTA processing , it will cause OTA fail.

So any suggestion to avoid this problem ?

**Comment by superm1** [comment, marked as answer]
Can you configure firmware to report a need to a retry in case this happens and then from code side do it again?

**Reply by jyu99** [reply]
We want to handle the OTA response , so we skip the other report id ,only accept the  OTA input report id. Keyboard report e,g normal key report length is 6 bytes , so our plugin can receive it  successfully and then  skip it . It's fine, when user input the media key e.g volume up or mute during the OTA processing  those report are 2 bytes, the  `fu_udev_device_pread_full ` will return false beacuse the `FU_PXI_DEVICE_NOTTFY_RET_LEN + 1` parameter. Finally the whole OTA result is fail

**Reply by superm1** [reply]
I feel like this needs to be changed either at firmware or driver level then to ignore sending these reports while in upgrade mode.
